### ISSUE TITLE
 Namespace '"stream"' has no exported member 'Stream'.   

### ISSUE THREAD

**Issue by whitecuteXu** [exceljs/exceljs#1191]
1337:40 Namespace '"events"' has no exported member 'EventEmitter'.
    1335 |       * Create input stream for reading
    1336 |       */
  > 1337 |      createInputStream(): import('events').EventEmitter;
         |                                          

1352:33 Namespace '"stream"' has no exported member 'Stream'.
    1350 |       * write to a stream
    1351 |       */
  > 1352 |      write(stream: import('stream').Stream, options?: Partial<CsvWriteOptions>): Promise<void>;
         |                                     ^
    1353 | }

**Comment by Alanscut** [comment]
@whitecuteXu  Please verify that ``node_modules`` already has ``@ types/node``** library?

**Comment by Alanscut** [comment]
Thanks for opening this request.
Please re-open if you still have issues.
Best of luck.

**Comment by passee** [comment]
Hello i am also getting the error

> Could not find a declaration file for module 'stream'. '/node_modules/stream/index.js' implicitly has an 'any' type.

Im using Angular v16
Component.ts:
`import * as ExcelJS from "exceljs";`
ts.config:
`"types": [
        "node"
    ],`

package.json:
`"devDependencies": {
    "@types/node": "^20.8.7",
  }`

i cant get it to work

**Comment by vzakharov-rxnt** [comment]
The fix did not work for me, same error. Angular 16, Devextreme grid.
- https://js.devexpress.com/Angular/Demos/WidgetsGallery/Demo/DataGrid/ExcelJSOverview/MaterialBlueLight/

**Comment by ansiboy** [comment]
Here is a solution, append a file named declare.d.ts, and content is like below:

```ts
declare module 'stream' {
    export type Stream = any;
}

declare module 'events' {
    export type EventEmitter = any;
}

declare namespace NodeJS {
    export type TypedArray = any;
}
```

**Comment by gustavo-reyes-garcia** [comment]
> Here is a solution, append a file named declare.d.ts, and content is like below:
> 
> ```ts
> declare module 'stream' {
>     export type Stream = any;
> }
> 
> declare module 'events' {
>     export type EventEmitter = any;
> }
> 
> declare namespace NodeJS {
>     export type TypedArray = any;
> }
> ```

It works like a charm, thank you so much 👍

**Comment by sayantan-git1** [comment]
> Here is a solution, append a file named declare.d.ts, and content is like below:
> 
> ```ts
> declare module 'stream' {
>     export type Stream = any;
> }
> 
> declare module 'events' {
>     export type EventEmitter = any;
> }
> 
> declare namespace NodeJS {
>     export type TypedArray = any;
> }
> ```

Where do i need to store this file and what should the path be for angular v15, exceljs @4.3.0?

**Comment by Ti-zum** [comment]
please if you can provide an explanation to the solution you have found, I've explored many solutions and none of them seems to work for my case i already working on existing project 

@ansiboy  please can you explain how you did it so i can try to do the same and thank you in advance ;) =)

working under angular 16 and I've got this:

`
Error: node_modules/exceljs/index.d.ts:1484:22 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

1484  read(stream: import('stream').Stream, options?: Partial<XlsxReadOptions>): Promise<Workbook>;
                          ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:1505:23 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

1505  write(stream: import('stream').Stream, options?: Partial<XlsxWriteOptions>): Promise<void>;
                           ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:1592:22 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

1592  read(stream: import('stream').Stream, options?: Partial<CsvReadOptions>): Promise<Worksheet>;
                          ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:1597:63 - error TS7016: Could not find a declaration file for module 'events'. 'C:/Users/Taied CHARGUI/Desktop/Zum-portal/zp_frontend/node_modules/events/events.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/events` if it exists or add a new declaration (.d.ts) file containing `declare module 'events';`

1597  createInputStream(options?: Partial<CsvReadOptions>): import('events').EventEmitter;
                                                                   ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:1612:23 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

1612  write(stream: import('stream').Stream, options?: Partial<CsvWriteOptions>): Promise<void>;
                           ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:1916:19 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

1916    stream: import('stream').Stream;
                       ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:2013:39 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

2013    constructor(input: string | import('stream').Stream, options: Partial<WorkbookStreamReaderOptions>);
                                           ~~~~~~~~


Error: node_modules/exceljs/index.d.ts:2022:18 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.

2022    entry: import('stream').Stream;`

**Comment by RuMont** [comment]
> ```ts
> declare module 'stream' {
>     export type Stream = any;
> }
> 
> declare module 'events' {
>     export type EventEmitter = any;
> }
> 
> declare namespace NodeJS {
>     export type TypedArray = any;
> }
> ```

Maybe a bit late but if you see in your tsconfig.app.json:

```json
"include": [
    "src/**/*.d.ts"
  ]
```

this will make every declaration file (*.d.ts) you add inside the src app be used in your project.

If you do not have these lines in your tsconfig.app.json try to add it and see. I fixed the error in Angular v17.3.9 with the declarations.
